subsquid/src/pabloProcessor.ts

```diff
@@ -0,0 +1,299 @@
+    console.info('processing SwappedEvent', event);
+    const swappedEvt = getSwappedEvent(event);
+    const who = ss58.codec("picasso").encode(swappedEvt.who);
+    const pool = await get(ctx.store, PabloPool, swappedEvt.poolId.toString());
+    // only set values if the owner was missing, i.e a new pool
+    if (pool != undefined) {
+        const isReverse: boolean = pool.quoteAssetId != swappedEvt.quoteAsset;
+        const timestamp = BigInt(new Date().getTime());
+        pool.transactionCount += 1;
+        pool.calculatedTimestamp = timestamp;
+        pool.blockNumber = BigInt(ctx.block.height);
+        // find baseAsset: Following is only valid for dual asset pools
+        const baseAsset = pool.poolAssets
+            .find((asset) => asset.assetId != pool.quoteAssetId);
+        if (baseAsset == undefined) {
+            throw new Error('baseAsset not found');
+        }
+        // find quoteAsset
+        const quoteAsset = pool.poolAssets
+            .find((asset) => asset.assetId == pool.quoteAssetId);
+        if (quoteAsset == undefined) {
+            throw new Error('quoteAsset not found');
+        }
+        if (isReverse) {
+            console.debug('Reverse swap');
+            // volume
+            pool.totalVolume = Big(pool.totalVolume).add(Big(swappedEvt.baseAmount.toString())).toString();
+            baseAsset.totalVolume += swappedEvt.quoteAmount;
+            quoteAsset.totalVolume += swappedEvt.baseAmount;
+
+            // liquidity
+            pool.totalLiquidity = Big(pool.totalLiquidity)
+                // fees TODO is this correct?
+                .sub(Big(swappedEvt.fee.toString()))
+                .toString();
+            // for reverse exchange "default quote" (included as the base amount in the evt) amount leaves the pool
+            baseAsset.totalLiquidity += swappedEvt.quoteAmount;
+            quoteAsset.totalLiquidity -= swappedEvt.baseAmount;
+            quoteAsset.totalLiquidity -= swappedEvt.fee;
+         } else {
+            console.debug('Normal swap');
+            // volume
+            pool.totalVolume = Big(pool.totalVolume).add(Big(swappedEvt.quoteAmount.toString())).toString();
+            baseAsset.totalVolume += swappedEvt.baseAmount;
+            quoteAsset.totalVolume += swappedEvt.quoteAmount;
+
+            // liquidity
+            pool.totalLiquidity = Big(pool.totalLiquidity)
+                // fees TODO is this correct?
+                .sub(
+                    // calculated the quote amount based on the exchange rate as the fees are in the base asset
+                    Big(swappedEvt.quoteAmount.toString())
+                        .div(Big(swappedEvt.baseAmount.toString()))
+                        .mul(Big(swappedEvt.fee.toString())))
+                .toString();
+            // for normal exchange "default quote" amount gets into the pool
+            baseAsset.totalLiquidity -= swappedEvt.baseAmount;
```

Comment:
Same as above.

Reply:
I think you possibly meant to comment on fees?

Reply:
yes.